api_app/core/views.py

```diff
@@ -59,14 +61,32 @@ def perform_kill(report: AbstractReport):
         performs kill
          override for callbacks after kill operation
         """
+        from api_app.models import Job
+
         # kill celery task
         celery_app.control.revoke(report.task_id, terminate=True)
         # update report
         report.status = AbstractReport.Status.KILLED
         report.save(update_fields=["status"])
+        # clean up job
 
-    def perform_retry(self, report: AbstractReport):
-        raise NotImplementedError()
+        job = Job.objects.get(pk=report.job.pk)
+        job.set_final_status()
+
+    @staticmethod
+    def perform_retry(report: AbstractReport):
+        signature = report.config.get_signature(
```

Comment:
can we add a line of log?

Reply:
The log is inside the view that call this method. I think that it is enough